src/shims/fs.rs

```diff
@@ -1062,6 +1062,36 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             this.handle_not_found()
         }
     }
+
+    fn ftruncate64(
+        &mut self, fd_op: OpTy<'tcx, Tag>,
+        length_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        this.check_no_isolation("ftruncate64")?;
+
+        let fd = this.read_scalar(fd_op)?.to_i32()?;
+        let length = this.read_scalar(length_op)?.to_i64()?;
+        if let Some(FileHandle { file, writable }) = this.machine.file_handler.handles.get_mut(&fd) {
+            if *writable {
+                if let Ok(length) = length.try_into() {
+                    let result = file.set_len(length);
+                    this.try_unwrap_io_result(result.map(|_| 0i32))
+                } else {
+                    let einval = this.eval_libc("EINVAL")?;
+                    this.set_last_error(einval)?;
+                    Ok(-1)
+                }
+            } else {
+                let einval = this.eval_libc("EINVAL")?;
```

Comment:
```suggestion
                // The file is not writeable.
                let einval = this.eval_libc("EINVAL")?;
```
There's enough `else`'s here that it's not always clear which branch we are in.

Also in the future we might want a method that just takes a string, returns `Ok(-1)`, and does the `set_last_error` stuff -- I think we have these 3 lines repeated plenty of times in this file.^^